### PR TITLE
Possessed Daemonic Swords for Cult

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Introduces possessed eldritch longswords for cult.

A possessed longsword is created by:

1: Placing the blade upon a sacrifice rune, and making a valid sacrifice (human/cyborg) on that rune. Mindless sacrifices will work (yes you can use monkey humans, but I figure this is fine as it's enough of a PITA).
2: Using the eldritch longsword while within 1 tile of an empowering rune on the station to begin the ritual of daemonic binding; this causes blood loss and bleeding.
3: This will prompt a ghost to possess the blade. Upon accepting the ghost will choose the blade's name and one of 15 unique enchantments to give it; it will also have access to cult chat:

"Blood" //Always causes heavy bleeding. Drains 50u Blood on each hit.
"Life Eating" //Heals attacker's brute damage by 40% of brute damage dealt.
"Corruption" //Deals 20 brute and injects 3U Unholy Water. Traps the victim in a soulstone if it delivers the killing blow.
"Pestilence" //Deals 20 brute and 10 toxin damage. Inflicts Unholy Wasting on a 30 second cooldown.
"Famine" //Deals 20 brute and 10 toxin damage. Massively drains nutrition on hit and injects 20U Lipolicide.
"Hellfire" //Deals burn damage instead of brute; inflicts burn stacks and lights on fire. Injects 3U phlogiston.
"Hellfrost" // Deals burn damage instead of brute; massively reduces target temperature and injects 10U Frost Oil
"Chaos" // Deals 2-80 damage of a random type (brute, burn, toxin); bell curve/gaussian distribution with 2 damage rolls. If both rolls have the same number, it shoots a bolt of chaos at the target.
"Madness" // Deals 20 Brute Damage, 20 Brain damage and inflicts Confusion and Hallucination stacks.
"Agony" //Deals 20 Brute, 35 Stamina damage on hit, with a stun baton stagger effect.
"Piercing" //+3 damage. Ignores armor.
"Warding" //Has a 50% melee and ranged deflection chance
"Devouring" //Deals 45 Brute, 5 Brute to the user.
"Returning" //Faster throw speed. 19 Brute damage on throw. Boomerangs back to the owner after being thrown.
"Brutality" //Deals double damage and imposes a big knock back on a 30 second cooldown.
"Destruction" //Deals triple damage to objects, double to silicons, can break walls on a cooldown.

Possessed swords can be purged by the Chaplain using his Bible while it is on ground blessed by holy water. When purged, the daemon inside will be released as a Cult Shade (it's too strong to be purified) and the sword will disintegrate into dust.

## Why It's Good For The Game

Adds a fun way to get players back into the game, while expanding the strategies available to the Cult and giving them entertaining new tools to play with.

## Changelog
:cl:
add: Added daemonic possession to cult eldritch longswords. After being empowered by a sacrifice, a sword can be used at an Empowering rune to summon a ghost to possess it and choose a name and enhancement for the weapon.
add: Possessed swords can be purged by the Chaplain using his Bible while it is on ground blessed by holy water. When purged, the daemon inside will be released as a Cult Shade and the sword will disintegrate into dust.
/:cl:

### CODE REVIEW

**Comment by Surrealaser** [comment]
If there's any concern about power creep, keep in mind that procuring everything you need to get one of these swords is pretty difficult.

File: code/FulpstationCode/fulp_cult/fulp_cult_swords.dm [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+	. = ..()
+	if(!iscultist(user))
+		return
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/H = user
+	switch(possessed) //Making this a switch in the event of future on-equip properties
+		if(CULT_HELLFIRE_SWORD)
+			REMOVE_TRAIT(H, TRAIT_NOFIRE, "hell_blade") //We lose immunity to combustion while we don't have the sword.
+			H.fire_stacks = 0 //Zero out any existing fire stacks.
```
Comment (remove32) [inline code/FulpstationCode/fulp_cult/fulp_cult_swords.dm:149]: if youre removing the trait why are you removing stacks? this seems like extra overhead

**Comment by remove32** [comment]
too much code to review in depth. im not sure if this is up to best practice but hey it compiles.

also you forgot to tick the new file in the dme

**Comment by remove32** [comment]
fucking bot

**Pull request closed without merging**